awsrequests/__init__.py

```diff
@@ -22,48 +23,40 @@ def emit(self, record):
             pass
 logging.getLogger(__name__).addHandler(NullHandler())
 
-class AwsRequester(object):
-    def __init__(self, region, access_key=None, secret_key=None, session_token=None, session_expires=None):
-        self.session_token = None
-        self.session_expires = None
-        self.region = region
-        if access_key is not None and secret_key is not None:
-            self.access_key = access_key
-            self.secret_key = secret_key
-            if session_token:
-                self.session_token = session_token
-                self.session_expires = session_expires
-        elif HAS_BOTO:
-            # hijack botocore's method.  probably fragile!
-            session = botocore.session.Session()
-            resolver = botocore.credentials.create_credential_resolver(session)
-            creds = resolver.load_credentials()
-            if creds is not None:
-                self.credentials = creds
-                self.access_key = creds.access_key
-                self.secret_key = creds.secret_key
-                self.session_token = creds.token
-            else:
-                raise EnvironmentError("could not find AWS creds anywhere!")
-        else:
-            raise EnvironmentError("could not find AWS creds (don't have boto3, so didn't look anywhere fancy)")
-
-    def assume_role(self, role_arn):
-        sts_client = boto3.client('sts',
-                                  region_name=self.region,
-                                  aws_access_key_id=self.access_key,
-                                  aws_secret_access_key=self.secret_key)
-        try:
-
-            temp_security_creds = sts_client.assume_role(RoleArn=role_arn,
-                                                         RoleSessionName="AwsBootstrapper"
-                                                         )
-            self.access_key = temp_security_creds["Credentials"]["AccessKeyId"]
-            self.secret_key = temp_security_creds["Credentials"]["SecretAccessKey"]
-            self.session_token = temp_security_creds["Credentials"]["SessionToken"]
-            self.session_expires = temp_security_creds["Credentials"]["Expiration"]
-        except botocore.exceptions.BotoCoreError as e:
-            raise Exception("error assuming role: {0}".format(e.args))
+class AwsRequester(AWSConnections):
+    def __init__(self, **kwargs):
+        """
+        **kwargs gets set to self.args as a dict
+
+        REQUIRED
+        :param: aws_asset = 'ec2', 'sts', 's3', 'dynamodb', any of the services
```

Comment:
So this library _can_ call to AWS's own APIs, but that's really not it's sweetspot.  Boto is so much better at that.  This is really just to give a wrapper for the requests (http://docs.python-requests.org/en/master/) library, for the primary purpose of talking to your own API Gateways that are protected by IAM.  This `aws_asset` is only ever used for construction of that client, which doesn't seem to get used anywhere.